CHANGELOG.rst

```diff
@@ -5,49 +5,48 @@ Changelog
 TBR
 ---
 
-* Scrapy uses the spider's ``parse()`` method when the :class:`scrapy.http.Request`
-  callback is set to ``None``.
+* Improved ``scrapy-poet`` behavior when ``spider.parse()`` method
+  arguments are annotated. Previously, they were causing unnecessary work
```

Comment:
We should mention that if it's only annotated by `DummyResponse` andanything else that **scrapy-poet** provides. If it's annotated with `scrapy.http.Response`, there are no issues with it. [Reference](https://github.com/scrapinghub/scrapy-poet/blob/master/tests/test_downloader.py#L492-L515).